test/memcached/cache_impl_test.go

```diff
@@ -1,594 +0,0 @@
-// Adapted from test/redis/cache_impl_test.go, with most test cases being the same
-// basic idea. TestMemcacheAdd() is unique to the memcache tests, since redis can create a new key
-// simply by incrementing it but memcached cannot. In memcache new keys need to be explicitly
-// added.
-package memcached_test
-
```

Comment:
please add tests for refactored code

Reply:
Hi sorry for very slow resp, will start creating unit test for the code